public/js/cin.js

```diff
@@ -0,0 +1,527 @@
+          onKeyUp=${handleSearch}
+          style="width: 25em;  max-width: 90%;"
+        />`,
+      ];
+      interviews.forEach((event) => {
+        // Spaces seem to break the search, so just yeet the space characters
+        if (
+          searchTerm.current == "" ||
+          event.interview.position.full_name
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1 ||
+          prettifyCandidates(event.interview.candidates)
+            .toLowerCase()
+            .replace(/\s/g, "")
+            .search(
+              escapeRegExp(searchTerm.current).toLowerCase().replace(/\s/g, "")
+            ) != -1
+        ) {
+          if (new Date(event.end_time).getTime() < Date.now()) {
+            var youtube = event.interview.youtube_id;
+            tmp.push(html`<${PastScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              youtubeStatus=${youtube != null
+                ? html`<a class="cin-text-2" href="javascript:void(0)"
+                    >Watch on YouTube</a
+                  >`
+                : "Available on YouTube Soon"}
+              callback=${updateYoutube}
+              youtubeID=${youtube}
+            />`);
+          } else if (new Date(event.start_time).getTime() > Date.now()) {
+            let time =
+              new Date(event.start_time).toLocaleTimeString().slice(0, -3) +
+              " - " +
+              new Date(event.end_time).toLocaleTimeString().slice(0, -3);
+            tmp.push(html`<${FutureScheduleCard}
+              time=${time}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+            />`);
+          } else {
+            let time =
+              "Now - " +
+              new Date(event.end_time).toLocaleTimeString().slice(0, -3);
+            tmp.push(html`<${LiveScheduleCard}
+              position=${event.interview.position.full_name}
+              candidate=${prettifyCandidates(event.interview.candidates)}
+              interviewer=${getInterviewers(event)}
+              time=${time}
+            />`);
+          }
+        }
+      });
+      if (tmp.length == 1) {
+        // No Interviews, Only Search
+        if (searchTerm.current == "") {
+          setSlots([html`<h2 class="text-center">Coming Soon...</h2>`]);
+        } else {
+          setSlots(
+            tmp.concat([
+              html`<br />
+                <h2 class="text-center">No Results</h2>`,
+            ])
+          );
+        }
+      } else {
+        setSlots(tmp);
+
+        if (youtubeTitle != "") {
+          setYoutubeVid(getLatestODVideo());
```

Comment:
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA!

This is global state, being used to drive component-local state. No. Burn. With fire.

Have `interviews` as state on a parent component, then pass it down. If that's too much ceremony, use context. Not global state.

There is no guarantee this will work. If it works, there is no guarantee it will continue to work.

Reply:
so

Reply:
It means that the output of the render function is non-deterministic, which breaks various assumptions that Preact makes. If it works, there’s no guarantee it won’t randomly break in hard to debug ways.